### PR TITLE
[AD-4056] Add CircleCI Config, Danger Checks, and CHANGELOG for Publishing Library

### PR DESCRIPTION
**Pull request by ckaethler**
## What does this PR do?
- Add CircleCI folder and file with publishing configuration
- Add `CHANGELOG` file to keep track of updates
- Add danger checks to yell at people to remember to update the changelog

## Resources

- [Jira ticket](https://getpantheon.atlassian.net/browse/AD-4056)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Use the latest 2.1 version of CircleCI pipeline process engine.
```
Comment (heyheyjose) [inline .circleci/config.yml:1]: this is just placeholder stuff?

Comment (ckaethler) [reply .circleci/config.yml]: Yes it came with the testing configuration file. I'm going through the tutorial on the CircleCI website right now to see how everything works

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        type: string
+    docker:
+      - image: quay.io/getpantheon/pantheon_node:<< parameters.image_tag >>
+        auth:
+          username: $QUAY_USER
+          password: $QUAY_PASSWD
+
+commands:
+  # TODO: Run danger checks
+  # TODO: Add CHANGELOG for updates
```
Comment (ckaethler) [inline .circleci/config.yml:16]: Adding these now whoopsie daisy

**Review by TeslaDethray — DISMISSED** [review]
I really appreciate the messages being in full sentences.

**Comment by ckaethler** [comment]
> I really appreciate the messages being in full sentences.

Thank you I try!!

**Review by BatyaFeldman — APPROVED** [review]

**Review by TeslaDethray — APPROVED** [review]

**Review by zzyou — APPROVED** [review]

**Pull request merged**